lib/Twig/Template.php

```diff
@@ -305,6 +305,33 @@ public function getBlocks()
     }
 
     /**
+     * Returns the template source code.
+     *
+     * @return string|null The template source code or null if it can't be retrieved
```

Comment:
It's not an error condition nor a logic mistake, just a case where the source code is not available. I could return an empty string if you prefer, but adding an exception will just force people to add try/catch boilerplate where a simple if would be enough